packages/react-query-devtools/src/devtools.tsx

```diff
@@ -1,343 +1,77 @@
-
-export { default as ReactQueryDevtoolsPanel } from './CachePanel/CachePanel'
-
+import { useQueryClient, onlineManager } from '@tanstack/react-query'
+import type {
+  DevtoolsButtonPosition,
+  DevtoolsPosition,
+} from '@tanstack/query-devtools'
+import { TanstackQueryDevtools } from '@tanstack/query-devtools'
+import React from 'react'
 export interface DevtoolsOptions {
   /**
    * Set this true if you want the dev tools to default to being open
    */
   initialIsOpen?: boolean
-  /**
-   * Use this to add props to the panel. For example, you can add className, style (merge and override default style), etc.
-   */
-  panelProps?: React.ComponentPropsWithoutRef<'div'>
-  /**
-   * Use this to add props to the close button. For example, you can add className, style (merge and override default style), onClick (extend default handler), etc.
-   */
-  closeButtonProps?: React.ComponentPropsWithoutRef<'button'>
-  /**
-   * Use this to add props to the toggle button. For example, you can add className, style (merge and override default style), onClick (extend default handler), etc.
-   */
-  toggleButtonProps?: React.ComponentPropsWithoutRef<'button'>
   /**
    * The position of the React Query logo to open and close the devtools panel.
    * Defaults to 'bottom-left'.
    */
-  position?: Corner
+  buttonPosition?: DevtoolsButtonPosition
   /**
    * The position of the React Query devtools panel.
    * Defaults to 'bottom'.
    */
-  panelPosition?: Side
-  /**
-   * Use this to render the devtools inside a different type of container element for a11y purposes.
-   * Any string which corresponds to a valid intrinsic JSX element is allowed.
-   * Defaults to 'aside'.
-   */
-  containerElement?: string | any
-  /**
-   * nonce for style element for CSP
-   */
-  styleNonce?: string
+  position?: DevtoolsPosition
+
   /**
    * Custom instance of QueryClient
    */
-  queryClient?: QueryClient
-  /**
-   * Use this so you can define custom errors that can be shown in the devtools.
-   */
-  errorTypes?: DevToolsErrorType[]
+  client?: QueryClient
 }
 
-export function ReactQueryDevtools({
-  initialIsOpen,
-  panelProps = {},
-  closeButtonProps = {},
-  toggleButtonProps = {},
-  position = 'bottom-left',
-  containerElement: Container = 'aside',
-  queryClient,
-  styleNonce,
-  panelPosition: initialPanelPosition = 'bottom',
-  errorTypes = [],
-}: DevtoolsOptions): React.ReactElement | null {
-  const rootRef = React.useRef<HTMLDivElement>(null)
-  const panelRef = React.useRef<HTMLDivElement>(null)
-  const [isOpen, setIsOpen] = useLocalStorage(
-    'reactQueryDevtoolsOpen',
-    initialIsOpen,
+export function ReactQueryDevtools(
+  props: DevtoolsOptions,
+): React.ReactElement | null {
+  const queryClient = useQueryClient()
+  const ref = useRef<HTMLDivElement>(null)
+  const { buttonPosition, position, initialIsOpen } = props
+  const [devtools] = useState(
+    new TanstackQueryDevtools({
```

Comment:
this should probably be a function:

```
useState(() => new TanstackQueryDevtools(...))
```

Reply:
Ah my bad! My brain is breaking switching my react brain on and off 😆